src/client/opamAdminRepoUpgrade.ml

```diff
@@ -372,6 +372,12 @@ let do_upgrade repo_root =
         "CAML_LD_LIBRARY_PATH", PlusEq, "%{lib}%/stublibs", None;
         "OCAML_TOPLEVEL_PATH", Eq, "%{toplevel}%", None;
       ] |>
+      O.with_extensions
+        (OpamStd.String.Map.singleton OpamFile.OPAM.rewrite_xfield
+           (OpamTypesBase.nullify_pos (OpamParserTypes.FullPos.Bool true))) |>
+      O.with_available
+        (FOp (FIdent ([], OpamVariable.of_string "opam-version", None),
+              `Geq, FString "2.2")) |>
```

Comment:
should we also have that addition for 1.2 comp -> 2.0 opam files ? see https://github.com/ocaml/opam/blob/8a23fc031b80abcaaf0e8c96059d02eea60d3fc6/src/state/opamFormatUpgrade.ml#L1396-L1398

Reply:
from dev meeting: to remove for ocaml package on opam admin upgrade, defined variables are package/switch variables.

Reply:
Finally, i've added an `x-allow-rewrite: false` to avoid new lint warning.